### ISSUE TITLE
use var instead of explicit type incorrectly reported in non-nullable out variable declaration

### ISSUE THREAD

**Issue by YairHalberstadt** [dotnet/roslyn#38743]
**Version Used**: VS 16.3 preview 1

**Steps to Reproduce**:

Set code style preferences to prefer var over explicit type.

Compile the following:

```csharp
#nullable enable
using System.Collections.Generic;

public class C {
    public void M() {
        var dic = new Dictionary<string, string>
        { {"key", "value"}};
        
        dic.TryGetValue("key", out string value);
    }
}
```
**Expected Behavior**:

No message to prefer var over explicit type, since `string` is different to `string?` 

**Actual Behavior**:

message to prefer var over explicit type.

**Comment by jasonmalinowski** [comment]
@jinujoseph What's the thing needing design review here? This looks like an outright bug where "use var" needs to also consider top-level nullability; it's a quick fix.

**Comment by jasonmalinowski** [comment]
@YairHalberstadt I'm guessing you meant to say 16.4 Preview 1 for the version used?

**Comment by YairHalberstadt** [comment]
@JasonMalinowski

I meant 16.3

I hadn't updated for a while at the time

**Comment by jinujoseph** [comment]
thanks @jasonmalinowski, will remove the tag..(actually forgot the reason to tag for design review)

**Comment by jasonmalinowski** [comment]
Still repros in 16.3 RTM.

**Comment by ryzngard** [comment]
Verified repro in 16.3. Does not repro in 16.4 preview 2 when using the following settings in editorconfig 

```
csharp_style_var_for_built_in_types = true:warning
csharp_style_var_when_type_is_apparent = true:warning
csharp_style_var_elsewhere = true:warning
```
